cardano-crypto-class/src/Cardano/Crypto/VRF/Class.hs

```diff
@@ -111,7 +109,7 @@ class ( Typeable v
   type Signable c = Empty
 
   evalVRF
-    :: (MonadRandom m, HasCallStack, Signable v a)
+    :: (Monad m, HasCallStack, Signable v a)
```

Comment:
Can we eliminate `Monad` and just make it pure? (and hence same for `evalCertified`)

Reply:
Well, we could demand that it be entirely pure, but then we can't error out anymore.

Reply:
Unless we throw from pure code, of course.

Reply:
OK, done. The Praos impl now just uses `error` to fail when it finds invalid keys.